pre-docs/inventory_version2.md

```diff
@@ -180,6 +180,77 @@ groups:
             hostname: facts.networking.interfaces.en0.ipaddress
 ```
 
+#### Task
+
+The Task plugin will run a task on `localhost` to lookup configuration
+information or target lists for inventory..
+
+For both use cases the plugin accepts two keys:
+* `task`: The task to run.
+* `parameters`: The parameters to pass to the task.
+
+example:
+
+```
+---
+version: 2
+targets:
+  - _plugin: task
+    task: my_json_file::targets
+    parameters:
+      # These parameters are specific to the task
+      file: /etc/targets/data.json
+      environment: production
+      app: my_app
+config:
+  ssh:
+    password:
+      _plugin: task
+      task: my_db::secret_lookup
+      parameters:
+        # These parameters are task specific
+        key: ssh_password
+```
+
+##### Inventory Config Tasks
+
+To lookup configuration information for the inventory return a `config` key in
```

Comment:
"To look up connection configuration return a hash or JSON object with a `config` key from the task. The hash should be the only thing output by the task. The value of `config` can be any form of data that is appropriate for the
specific location in config. For example `host-key-check` should be a boolean value, `run-as` should be a string, and `run-as-command` should be an array, so a task that sets these should return:
```
{
"config": {
  "host-key-check": true,
  "run-as": "sudo-user",
  "run-as-command": [ "sudo", "-k", "-S", "-E", "-u", "user", "-p", "password"] 
}
}
```